### DISCUSSION TITLE
Are the SELinux instructions buggy?

### DISCUSSION THREAD

**Discussion by webartifex** [rustdesk/rustdesk#7297] [General]
Hi there,

I am on Fedora 39 and RustDesk seems to work fine when started manually.

However, when I go through the SELinux instructions, I get the following error when running `make`:

```
make -f /usr/share/selinux/devel/Makefile rustdesk.pp
make[1]: Entering directory '/home/alexander/rustdesk-selinux-1.0'
Compiling targeted rustdesk module
rustdesk.te:412:ERROR 'unknown role system' at token ';' on line 4633:
 	
role system types rustdesk_t;
/usr/bin/checkmodule:  error(s) encountered while parsing configuration
make[1]: *** [/usr/share/selinux/devel/include/Makefile:157: tmp/rustdesk.mod] Error 1
make[1]: Leaving directory '/home/alexander/rustdesk-selinux-1.0'
make: *** [Makefile:13: rustdesk.pp] Error 2
```

What am I missing?

**Comment by rustdesk** [comment]
@fufesou

**Comment by webartifex** [comment]
If you tell me how I can help you fix this, I will try my best :)

**Comment by fufesou** [comment]
Hi, thanks for your feedback.

I've tried the instructions here https://rustdesk.com/docs/en/client/linux/selinux/#:~:text=00%3A02%20rustdesk-,Run%3A,-%24%20cd%20rustdesk

It works fine in a new VM Fedora39.

Can you please run `sed -n '412p' rustdesk.te` ?

It seems the code is wrong in your file.

I can see the same error if the line is `role system types rustdesk_t;`

But the correct line should be  `role system_r types rustdesk_t;`

**Comment by webartifex** [comment]
The output is `role system types rustdesk_t;`

I simply copied and pasted the contents from the tutorial. Maybe there is a typo?
